src/Contributor/ContributorEnrichedRepository.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Contributor;
+
+use CultuurNet\UDB3\Model\ValueObject\Identity\UUID;
+use CultuurNet\UDB3\ReadModel\DocumentRepository;
+use CultuurNet\UDB3\ReadModel\DocumentRepositoryDecorator;
+use CultuurNet\UDB3\ReadModel\JsonDocument;
+use CultuurNet\UDB3\Role\ValueObjects\Permission;
+use CultuurNet\UDB3\Security\Permission\PermissionVoter;
+use CultuurNet\UDB3\StringLiteral;
+
+final class ContributorEnrichedRepository extends DocumentRepositoryDecorator
+{
+    private ContributorRepository $contributorRepository;
+
+    private PermissionVoter $permissionVoter;
+
+    private ?string $currentUserId;
+
+    public function __construct(
+        ContributorRepository $contributorRepository,
+        DocumentRepository $documentRepository,
+        PermissionVoter $permissionVoter,
+        ?string $currentUserId
+    ) {
+        parent::__construct($documentRepository);
+        $this->contributorRepository = $contributorRepository;
+        $this->permissionVoter = $permissionVoter;
+        $this->currentUserId = $currentUserId;
+    }
+
+    public function fetch(string $id, bool $includeMetadata = false): JsonDocument
+    {
+        $jsonDocument = parent::fetch($id, $includeMetadata);
+
+        if ($this->hasPermission($jsonDocument->getId())) {
+            $jsonDocument = $this->enrich($jsonDocument);
+        } else {
+            $jsonDocument = $jsonDocument->applyAssoc(
+                function (array $json) {
```

Comment:
For some reason this is necessary for places. If you are unknown or unauthorized you get an empty array for contributors in the JSON Projection of a `Place`. This does not happen with the Projection of an `Event` or an `Organizer`.

Reply:
I would search for the root cause of this issue. Do you get it inside a unit test or when running the acceptance tests or when doing a manual test?

Reply:
This was the problem: 
`            function (array $body) use ($contributors) {
                    $body['contributors'] = $contributors;
                return $body;
            }`
            Solution was to add a check to see if the `array` is empty or not.

Reply:
As discussed through Slack the issue is a little bit more subtle than this. When a projection gets updated by an event the existing projection gets `fetched` from Redis, the change related to the event gets applied, and the JSON gets `saved` again.

With this extra enricher to add contributors the `fetch` method will always return JSON with contributors when authenticated and they will get saved in Redis. So the next time the JSON projection is `fetched` the contributors are still on the JSON.

For other enriched data like popularity, this will not happen because the popularity scores are only added when `includeMetadata` is true. That flag is always false inside the projectors and only true inside the HTTP request handlers.

Proposed solution is to only embed the contributors when `includeMetadata` is true.

Reply:
Proposed solution is to only embed the contributors when `includeMetadata` is true will not work cause `Organizer` endpoint does not have `metadata`

Reply:
solution:`contributors` will be shown in the JsonBody and are removed in `save()`.
Unit & Acceptance tests have been added to verify this behaviour.